### ISSUE TITLE
BUG: Wrong (?) documentation in geopandas.sindex.SpatialIndex.query

### ISSUE THREAD

**Issue by tqa236** [geopandas/geopandas#2668]
- [x] I have checked that this issue has not already been reported.

I tried to search with keywords like `sindex`, `query` but found no similar results

- [x] I have confirmed this bug exists on the latest version of geopandas.

On the latest version of the documentation

- [ ] (optional) I have confirmed this bug exists on the main branch of geopandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
import geopandas as gpd
from shapely.geometry import Point, Polygon

gdf = gpd.GeoDataFrame(
    geometry=[
        Point([0.25, 0.25]),
        Point([0.75, 0.75]),
        Polygon([[-1, 0], [-1, 3], [1, 3], [-1, 0]]),
        Polygon([[-1, 0], [-1, 3], [1, 3], [1, 0], [-1, 0]]),
    ],
    crs="epsg:4326",
)

extent = gpd.GeoDataFrame(
    geometry=[Polygon([[0, 0], [0, 1], [1, 0], [0, 0]])],
    crs="epsg:4326",
)

gdf[gdf.intersects(extent.geometry.unary_union)]
# Point([0.25, 0.25])
# Polygon([[-1, 0], [-1, 3], [1, 3], [1, 0], [-1, 0]])

indices = gdf.sindex.query(extent.geometry.unary_union, predicate="intersects")
gdf.iloc[indices, :]
# Point([0.25, 0.25])
# Polygon([[-1, 0], [-1, 3], [1, 3], [1, 0], [-1, 0]])
```

#### Problem description

The documentation of [`geopandas.sindex.SpatialIndex.query`](https://geopandas.org/en/latest/docs/reference/api/geopandas.sindex.SpatialIndex.query.html) says 

> Return the index of all geometries in the tree with extents that intersect the envelope of the input geometry.

"Envelope" has a precise meaning in `geopandas`: [source](https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoSeries.envelope.html#geopandas.GeoSeries.envelope)

> The envelope of a geometry is the bounding rectangle. That is, the point or smallest rectangular polygon (with sides parallel to the coordinate axes) that contains the geometry.

However, from the code snippet above, it seems that the actual geometry is used, not the envelope, otherwise, the triangle at index 2 is selected. 

#### Expected Output

Remove the envelope part from the documentation

#### Output of ``geopandas.show_versions()``

<details>

[paste the output of ``geopandas.show_versions()`` here leaving a blank line after the details tag]
SYSTEM INFO
-----------
python     : 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]
executable : /home/anhtrinh/virtual_env/tmp/bin/python3
machine    : Linux-5.15.79.1-microsoft-standard-WSL2-x86_64-with-glibc2.35

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : 3.10.2
GEOS lib   : /usr/lib/x86_64-linux-gnu/libgeos_c.so
GDAL       : 3.4.3
GDAL data dir: /home/anhtrinh/virtual_env/tmp/lib/python3.10/site-packages/fiona/gdal_data
PROJ       : 9.1.0
PROJ data dir: /home/anhtrinh/virtual_env/tmp/lib/python3.10/site-packages/pyproj/proj_dir/share/proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 0.12.1
numpy      : 1.23.5
pandas     : 1.5.2
pyproj     : 3.4.0
shapely    : 1.8.4
fiona      : 1.8.22
geoalchemy2: None
geopy      : 2.3.0
matplotlib : 3.6.2
mapclassify: 2.4.3
pygeos     : 0.13
pyogrio    : v0.4.2
psycopg2   : None
pyarrow    : 10.0.1
rtree      : 1.0.1

</details>

**Comment by martinfleis** [comment]
The actual geometry is used because you have specified predicate. The the index checks the actual operation after filtering using the envelope. If you set it to None, you'll get what you expect. But it is true that we should try to clarify that in docs. Thanks!

**Comment by tqa236** [comment]
@martinfleis  Thanks a lot for the quick reply. It's clear to me now.

**Comment by mahdin75** [comment]
Is there any work on the current issue?

**Comment by martinfleis** [comment]
@mahdin75 I don't think so but check the current state of the docstring. It may be better than it used to be.
